src/openrct2/ride/Ride.cpp

```diff
@@ -4119,13 +4111,11 @@ static bool ride_check_start_and_end_is_station(CoordsXYE* input)
 static void ride_set_boat_hire_return_point(Ride* ride, CoordsXYE* startElement)
 {
     int32_t trackType = -1;
-    int32_t returnX = startElement->x;
-    int32_t returnY = startElement->y;
-    int32_t startX = returnX;
-    int32_t startY = returnY;
-    TileElement* returnTrackElement = startElement->element;
+    auto returnPos = *startElement;
+    int32_t startX = returnPos.x;
```

Comment:
remove these variables only trackBeginEnd should be defined here.

Reply:
The `while` alters the `returnPos` value. If I remove it and write to `startElement` instead, I'll be modifying a pointer which I don't own (and that the caller actually uses later). Is this really intended?